Created a survey result page that doesn't require password

loisaleghe opened the pull request:
This is an optional feature but if the user doesn't mind the public seeing the results they can choose this option if they do they can say no too.

loisaleghe committed:
Update storeAnswers.js

I removed redundant comments as suggested

loisaleghe committed:
creating this pull so I can merge with master before creating test

loisaleghe committed:
Merge branch 'master' of https://github.com/oyenmwen/SurveyMonkey into Lois-branch

loisaleghe committed:
created a test case for the linkedquestion to answer

Although now the test says pass but its incompleted hence why i'm not merging it to master yet

loisaleghe committed:
Merge branch 'Lois-branch' of https://github.com/oyenmwen/SurveyMonkey into Lois-branch

loisaleghe committed:
Merge branch 'master' of https://github.com/oyenmwen/SurveyMonkey into Lois-branch

 Conflicts:
	src/main/resources/public/storeAnswers.js
	src/test/java/com/surveyMonkey/AdminControllerTest.java

loisaleghe committed:
created survey results but without passwords

Gives user the option to display to public or not. If they choose not to then they would require a password to view result

loisaleghe commented:
test has not been added yet

JulianMendoza commented:
You could probably just reuse the existing code to display the results with a toggle variable but if this works its okay! Please add tests to this pull request, it's part of the CI/CD pipeline and it will definitely be something that the TA is looking for. Other than that, looks great!

loisaleghe commented:
> You could probably just reuse the existing code to display the results with a toggle variable but if this works its okay! Please add tests to this pull request, it's part of the CI/CD pipeline and it will definitely be something that the TA is looking for. Other than that, looks great!

okay, I will do that right away

loisaleghe commented:
> You could probably just reuse the existing code to display the results with a toggle variable but if this works its okay! Please add tests to this pull request, it's part of the CI/CD pipeline and it will definitely be something that the TA is looking for. Other than that, looks great!

Also it is true that I could have used existing code already present but since the existing required both title and password to be true I wanted to create one without password that would be separate in case the existing one breaks it doesn't fully affect this , it also felt more straight forward that way but for future development to reduce stress I'll probably do what you suggested!!

loisaleghe committed:
Update AdminControllerTest.java

added test cases or the survey results page that does not require passwords

loisaleghe commented:
the test has been added now

oyenmwen commented on src/main/java/com/surveyMonkey/controllers/AdminController.java:
```diff
@@ -135,4 +155,21 @@ public String deleteSurvey(@RequestParam("surveyCode") String surveyCode, @Reque
         }
         return "deleteSurvey";
     }
+    @PostMapping({"/deleteSurveyWithoutPswrd"})
```
What is the purpose of delete survey without a password feature? Shouldn't be an admin/authorized only capability ? Shouldnt only the owner be allowed to delete the survey ?

loisaleghe replied:
> What is the purpose of delete survey without a password feature? Shouldn't be an admin/authorized only capability ? Shouldnt only the owner be allowed to delete the survey ?

You are right only the owner should be able to delete survey, I didn't think of that when I was writing the code, good catch, thank you!

loisaleghe committed:
deleted the controller for deleting survey without password

Only the person who created the survey should be able to delete it not just anyone which was I mistake I made the first time

loisaleghe commented:
I have fixed it now

oyenmwen reviewed (approved):
Looks great Wendy good job
